test/integration/test-actions.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as sinon from 'sinon';
+import {AmpEvents} from '../../src/amp-events';
+import {
+  createFixtureIframe,
+  poll,
+} from '../../testing/iframe.js';
+
+describe.configure().retryOnSaucelabs().run('on="..."', () => {
+  let fixture;
+  let sandbox;
+
+  beforeEach(() => {
+    sandbox = sinon.sandbox.create();
+
+    return createFixtureIframe('test/fixtures/actions.html', 500).then(f => {
+      fixture = f;
+
+      // Wait for one <amp-img> element to load.
+      return fixture.awaitEvent(AmpEvents.LOAD_END, 1);
+    });
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  describe('"tap" event', () => {
```

Comment:
Maybe switch to the new `describes.integration`? I find tests written with that to be simpler and less flaky. No needs for things like `fixture.awaitEvent(AmpEvents.LOAD_END, 1)`

Reply:
That's strange since `describes.integration` is just bootstrapping an iframe behind the scenes. I'm not sure why AMP components would be guaranteed to be loaded by the time `onload` is fired.

Reply:
you are correct, it doesn't wait for load of individual elements. I have been using `poll + queryString` a lot to wait for things to load before asserting instead of `awaitEvent(AmpEvents.LOAD_END` (e.g. `poll( queryString amp-img img)`)

Is there a real need to wait to wait for amp-img to load? In this test I feel like it should just work even if amp-img is not layedout.

Reply:
Fair point. It needs to wait for the Actions service to initialize and install event handlers etc., but we don't have an `AmpEvent` for that so I'm using `LOAD_END` as a proxy. Users should only be able to perform actions once in-viewport components are loaded anyways.